flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/Fabric8FlinkKubeClient.java

```diff
@@ -279,7 +280,7 @@ public void eventReceived(Action action, Pod pod) {
 
 			@Override
 			public void onClose(KubernetesClientException e) {
-				LOG.error("The pods watcher is closing.", e);
+				throw new FlinkRuntimeException("The pods watcher should not be closed.", e);
```

Comment:
Could you explain how this is solving the problem @wangyang0918? Looking at the code, this should throw an exception in some `KubernetesClient` internal thread which does not guarantee that it is properly handled. I think it should rather trigger some behaviour in `KubernetesResourceManager` (e.g. trying to reconnect or in the worst case to fail with a descriptive error message).

Reply:
@tillrohrmann Thanks a lot for your quick review. Yes, you are right. Of cause, we could not just throw a `FlinkRuntimeException` directly. It needs to be handled more appropriately. When the pods watcher is closed exceptionally, it should be handled by the caller. In our case, the`KubernetesResourceManager` will start a new watch.

I was just fully occupied yesterday by some internal things and forgot to push the latest codes to this PR. :smile: